### ISSUE TITLE
Issue with new amazon SDK testing against S3Mock when deleting objects

### ISSUE THREAD

**Issue by spinscale** [findify/s3mock#71]
I ran into a problem with the new SDK and trying to delete objects, this is the logging HTTP dump from my test

```
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "POST /owntracks?delete HTTP/1.1[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "Host: localhost:12345[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "amz-sdk-invocation-id: d7c5488c-7256-340e-cfbb-32c547da49e4[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "amz-sdk-retry: 0/0/500[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "Authorization: AWS4-HMAC-SHA256 Credential=foo/20171022/us-east-1/s3/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-retry;content-length;content-md5;content-type;host;user-agent;x-amz-content-sha256;x-amz-date, Signature=bf727afacd579dfd6a27d23ba37e5ed87e3f709c8aca7cc9212f87fddea080b1[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "Content-MD5: RzqevkhpjQjuCXOin/jQEw==[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "Content-Type: application/xml[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "User-Agent: aws-sdk-java/2.0.0-preview-4 Mac_OS_X/10.12.6 Java_HotSpot(TM)_64-Bit_Server_VM/25.111-b14/1.8.0_111 scala/2.12.2 kotlin/1.1.51[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "x-amz-content-sha256: c1af73964eed3760be9468ffd2c01b9c537dc5b0735dba6e4487cd11cae3e8ea[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "X-Amz-Date: 20171022T190821Z[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "Content-Length: 211[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "Connection: Keep-Alive[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:72 - http-outgoing-0 >> "[\r][\n]"
2017-10-22 21:08:21 <> DEBUG org.apache.http.wire:86 - http-outgoing-0 >> "<?xml version="1.0" encoding="UTF-8"?><Delete xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Object><Key>data/2017-07-31/10:34.json</Key></Object><Object><Key>data/2017-07-31/16:23.json</Key></Object></Delete>"
2017-10-22 21:08:21 <> ERROR io.findify.s3mock.S3Mock:57 - method not implemented: POST http://localhost:12345/owntracks?delete
```

Just FYI: `owntracks` is the name of my bucket. Using `software.amazon.awssdk:s3:2.0.0-preview-4` to run against S3Mock.

Test setup code (sorry, it's kotlin)

```
                var api = S3Mock.Builder().withPort(12345).withInMemoryBackend().build()
                api.start()
                val s3Client : S3Client = S3Client.builder()
                        .endpointOverride(URI("http://localhost:12345")).region(Region.US_EAST_1)
                        .advancedConfiguration(S3AdvancedConfiguration.builder().pathStyleAccessEnabled(true).build())
                        .build()
```

failing code

```
        val deleteBuilder = DeleteObjectsRequest.builder().bucket(BUCKET)
        val ids = sortedObjects.map { ObjectIdentifier.builder().key(it.key()).build() }.toMutableList()
        deleteBuilder.delete(Delete.builder().objects(ids).build())
        val deleteObjectResponse = s3Client.deleteObjects(deleteBuilder.build())
```

Still hoping I did something stupid, that you might be better to spot than I do.

**Comment by shuttie** [comment]
Looks like that the root problem is not with the AWS SDK 2.0 itself, but with the "delete multiple objects" api call not being correctly implemented for some reason.

**Comment by shuttie** [comment]
Fixed in 0.2.4
